apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -44,63 +47,86 @@ const profile = {
 module.exports = class Filepanel extends ViewPlugin {
   constructor (appManager) {
     super(profile)
-    var self = this
-    self._components = {}
-    self._components.registry = globalRegistry
-    self._deps = {
-      fileProviders: self._components.registry.get('fileproviders').api,
-      fileManager: self._components.registry.get('filemanager').api,
-      config: self._components.registry.get('config').api
+    this._components = {}
+    this._components.registry = globalRegistry
+    this._deps = {
+      fileProviders: this._components.registry.get('fileproviders').api,
+      fileManager: this._components.registry.get('filemanager').api,
+      config: this._components.registry.get('config').api
     }
-
-    function createProvider (key, menuItems) {
-      return new FileExplorer(self._components.registry, self._deps.fileProviders[key], menuItems, self)
+    this.hideRemixdExplorer = true
+    this.remixdExplorer = {
+      hide: () => {
+        this.hideRemixdExplorer = true
+        this.renderComponent()
+      },
+      show: () => {
+        this.hideRemixdExplorer = false
+        this.renderComponent()
+      }
     }
-
-    var fileExplorer = createProvider('browser', ['createNewFile', 'publishToGist', canUpload ? 'uploadFile' : ''])
-    var fileSystemExplorer = createProvider('localhost')
-
-    self.remixdHandle = new RemixdHandle(fileSystemExplorer, self._deps.fileProviders.localhost, appManager)
-    self.gitHandle = new GitHandle()
-
-    const explorers = yo`
-      <div>
-        <div class="pl-2 ${css.treeview}" data-id="filePanelFileExplorerTree">${fileExplorer.init()}</div>
-        <div class="pl-2 filesystemexplorer ${css.treeview}">${fileSystemExplorer.init()}</div>
+    this.el = yo`
+      <div id="fileExplorerView">
       </div>
     `
 
-    function template () {
-      return yo`
-        <div class=${css.container}>
-          <div class="${css.fileexplorer}">           
-            <div class="${css.fileExplorerTree}">
-              ${explorers}
-            </div>
-          </div>
-        </div>
-      `
-    }
+    this.remixdHandle = new RemixdHandle(this.remixdExplorer, this._deps.fileProviders.localhost, appManager)
+    this.gitHandle = new GitHandle()
 
-    var event = new EventManager()
-    self.event = event
-    var element = template()
-    fileExplorer.ensureRoot()
-    self._deps.fileProviders.localhost.event.register('connecting', (event) => {
+    this.event = new EventManager()
+    // fileExplorer.ensureRoot()
+    this._deps.fileProviders.localhost.event.register('connecting', (event) => {
     })
 
-    self._deps.fileProviders.localhost.event.register('connected', (event) => {
-      fileSystemExplorer.show()
-    })
+    // this._deps.fileProviders.localhost.event.register('connected', (event) => {
```

Comment:
This is handled somewhere else or you don't want anymore these event to apply?

Reply:
They are still valid events. I will update my PR.